### ISSUE TITLE
Picture in showDialog not in the message sent (method = 'send')

### ISSUE THREAD

**Issue by Ziv-Barber** [jeduan/cordova-plugin-facebook4#160]
I set the 'picture' data member in the object that I passed to showDialog. I can see the image in the send dialog but a random image when reading the message been sent. If my link including '?' then there's no image at all.
I guess there's an undocumented data member for the image sent with the link.
BTW: I found another undocumented data member effecting 'send' dialogs: name (the caption of the link in the message).
If anyone managed to find more undocumented data members please share.
Known data members:
description: the message to send.
name: the header of the link.
link: the link to send.
picture: the image only in the send dialog.
caption: the header of the sent dialog (?).
Anyone managed to find more secret data members for showDialog?

**Comment by derulo** [comment]
I am having exactly the same issue.
The picture is included in the message content, however when reading the sent message, it shows the picture of the website that was sent as link. Also, the receiver actually does not receive any picture at all.

**Comment by Ziv-Barber** [comment]
After a lot of investigations we found out that on the link that you sharing you MUST add some open graph metadata:

&lt;meta property="og:url" content="https://your-site.com/" /&gt;
&lt;meta property="og:type" content="website" /&gt;
&lt;meta property="og:title" content="--- will be the link's header ---" /&gt;
&lt;meta property="og:description" content="--- will be the link's description ---" /&gt;
&lt;meta property="og:image" content="http://yoursite.com/your-image.png" /&gt;

Warnings:
1. og:url MUST be the SAME as the link that you passing as the 'link' parameter to shareDialog.
2. If your link including http get query ('?') then users will NOT see the picture!!!

**Comment by derulo** [comment]
Hi,
thanks for the super-quick answer! However, sharing of the link works using the href parameter, I need to share a picture which does not appear in the dialog window.
Thanks a lot!

**Comment by derulo** [comment]
Hi,
thanks for the super quick answer! However, is it possible to send over the picture that is actually attached to the message using the picture parameter, and not the og:image property of the link?
Thanks!

**Comment by Ziv-Barber** [comment]
The 'picture' parameter effecting the showDialog (user's will see it in the share dialog).
The 'og:image' meta is what the user that getting the message will see (the picture in the shared content).

**Comment by derulo** [comment]
OK, but I wanted to send a different picture each time (as set in the 'picture' parameter), not one static picture stored in the meta data of the website.
So if I get it right, the received message contains the data stored in the Meta data rather than what is passed in the showDialog parameters...

**Comment by Ziv-Barber** [comment]
Yes, you got it right + you can't use '?' in your link (or you'll not see any picture at all).
The solutions may be to use on your server a syntax something like this:
mysite.com/fb/the-code-of-the-picture-to-share/
the-code-of-the-picture-to-share = the code what picture to share.
then the server should create the meta data automatic from the-code-of-the-picture-to-share.
Not sure if it's working but you can try it.

**Comment by derulo** [comment]
Thanks for the hint, may try it out later :) Thanks for your help!!!
